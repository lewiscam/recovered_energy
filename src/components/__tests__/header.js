import React from "react"
import { shallow, mount, render } from "enzyme"
import Header from "../header"
import { useStaticQuery } from "gatsby"
describe("Header", () => {
  beforeEach(() => {
    useStaticQuery.mockImplementationOnce(() => ({
      allPrismicModel: {
        nodes: [
          {
            uid: "model1",
            data: {
              product_name: {
                text: "Model 1",
              },
            },
          },
        ],
      },
      allPrismicSystemCategory: {
        nodes: [
          {
            uid: "systemCat1",
            data: {
              system_category_title: {
                text: "System Category 1",
              },
            },
          },
        ],
      },
    }))
  })
  it("renders correctly", () => {
    const tree = render(<Header />)
    expect(tree).toMatchSnapshot()
  })
})
