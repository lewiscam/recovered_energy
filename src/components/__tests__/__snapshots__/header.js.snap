// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<header>
  <Menu
    primary={true}
  >
    <MenuItem>
      <mockConstructor
        to="/"
      >
        <Image />
      </mockConstructor>
    </MenuItem>
    <MenuMenu
      position="right"
    >
      <MenuItem>
        <mockConstructor
          to="/about-us"
        >
          About Us
        </mockConstructor>
      </MenuItem>
      <Dropdown
        additionLabel="Add "
        additionPosition="top"
        aria-label="Products"
        className="link item"
        closeOnBlur={true}
        closeOnEscape={true}
        deburr={false}
        icon="dropdown"
        minCharacters={1}
        noResultsMessage="No results found."
        openOnFocus={true}
        renderLabel={[Function]}
        searchInput="text"
        selectOnBlur={true}
        selectOnNavigation={true}
        text="Products"
        wrapSelection={true}
      >
        <DropdownMenu>
          <mockConstructor
            key="0"
            to="/products/model1"
          >
            <DropdownItem
              key="0"
            >
              Model 1
            </DropdownItem>
          </mockConstructor>
        </DropdownMenu>
      </Dropdown>
      <MenuItem>
        <mockConstructor
          to="/contact"
        >
          Contact Us
        </mockConstructor>
      </MenuItem>
      <Dropdown
        additionLabel="Add "
        additionPosition="top"
        aria-label="Systems"
        className="link item"
        closeOnBlur={true}
        closeOnEscape={true}
        deburr={false}
        icon="dropdown"
        minCharacters={1}
        noResultsMessage="No results found."
        openOnFocus={true}
        renderLabel={[Function]}
        searchInput="text"
        selectOnBlur={true}
        selectOnNavigation={true}
        text="Systems"
        wrapSelection={true}
      >
        <DropdownMenu>
          <mockConstructor
            key="0"
            state={
              Object {
                "item": Object {
                  "data": Object {
                    "system_category_title": Object {
                      "text": "System Category 1",
                    },
                  },
                  "uid": "systemCat1",
                },
              }
            }
            to="/systems/systemCat1"
          >
            <DropdownItem
              key="0"
            >
              System Category 1
            </DropdownItem>
          </mockConstructor>
        </DropdownMenu>
      </Dropdown>
    </MenuMenu>
  </Menu>
</header>
`;
